lib/credentials.js

```diff
@@ -148,3 +148,6 @@ AWS.Credentials = AWS.util.inherit({
     callback();
   }
 });
+
+AWS.util.addPromises(AWS.Credentials);
+AWS.util.addPromises(AWS.CredentialProviderChain);
```

Comment:
Due to the order the modules are loaded, moving that line to the `credential_provider_chain.js` file would throw an error. However, I discovered this is due to the fact that I'm loading `config.js` from inside the `AWS.util.addPromises()` (which is needed if I want to get the promise dependency via `AWS.config.getPromisesDependency()`. I can refactor that so that I can move this line to the right file.